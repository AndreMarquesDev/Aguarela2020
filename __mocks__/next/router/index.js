jest.mock('next/router', () => ({
    ...jest.requireActual('next/router'),
    useRouter() {
        return {
            route: '/',
            pathname: '',
            query: {
                language: 'pt',
            },
            asPath: '',
        };
    },
}));

module.exports = jest.requireMock('next/router');
