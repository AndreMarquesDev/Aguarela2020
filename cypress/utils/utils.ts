import { Locale } from '../../types/Locale';
import {
    menuIconButtonDataTestId,
    nukaCarouselNextButtonDataTestId,
} from '../../utils/dataTestIds';
import { TextsInterface, textsPt, textsEn } from '../../utils/texts';

export const getLocalizedTexts = (locale: Locale): TextsInterface => {
    if (locale === Locale.Pt) {
        return textsPt;
    }

    return textsEn;
};

export const openMenuMobile = (isMobile: boolean, isFirefox: boolean): void => {
    if (isMobile) {
        cy.getByDataTestId(menuIconButtonDataTestId).click({ force: isFirefox });
    }
};

export const clickNextArrowButtonIfMobile = (
    isMobile: boolean,
    container: string,
    containerTitle: string
): void => {
    const isFirefox = Cypress.isBrowser('firefox');

    if (isMobile) {
        cy.getByDataTestIdParent(container, nukaCarouselNextButtonDataTestId).click({
            force: isFirefox,
        });

        cy.getByText(container, containerTitle).scrollIntoView();
    }
};
