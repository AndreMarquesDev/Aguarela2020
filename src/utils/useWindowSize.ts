import { useEffect, useState } from 'react';

type WindowSize = {
    width: number;
    height: number;
};

export const useWindowSize = (): WindowSize => {
    const [windowSize, setWindowSize] = useState<WindowSize>({
        width: 0,
        height: 0,
    });

    useEffect(() => {
        const handleResize = (): void => {
            // eslint-disable-next-line react-hooks-extra/no-direct-set-state-in-use-effect
            setWindowSize({
                width: window.innerWidth,
                height: window.innerHeight,
            });
        };

        window.addEventListener('resize', handleResize);

        handleResize();

        return (): void => window.removeEventListener('resize', handleResize);
    }, []);

    return windowSize;
};

export enum Breakpoint {
    Desktop = 1280,
    Phablet = 1023,
    Tablet = 767,
    Mobile = 511,
}
