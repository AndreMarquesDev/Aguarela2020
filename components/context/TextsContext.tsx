import { createContext, Context } from 'react';
import { TextsInterface, textsPt } from '../../utils/texts';

export interface TextsContextProps {
    texts: TextsInterface;
}

export const TextsContext: Context<TextsContextProps> = createContext({
    texts: textsPt,
});
