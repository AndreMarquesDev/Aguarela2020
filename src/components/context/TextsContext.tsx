import type { Context } from 'react';
import type { TextsInterface } from '../../utils/texts';
import { createContext } from 'react';
import { textsPt } from '../../utils/texts';

export type TextsContextProps = {
    texts: TextsInterface;
};

export const TextsContext: Context<TextsContextProps> = createContext({
    texts: textsPt,
});
