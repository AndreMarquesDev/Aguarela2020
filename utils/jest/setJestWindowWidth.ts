import { Breakpoint } from '../useWindowSize';

export const setJestWindowWidth = (breakpoint: Breakpoint): void => {
    let width;

    if (breakpoint === Breakpoint.Mobile) {
        width = Breakpoint.Mobile - 1;
    } else if (breakpoint === Breakpoint.Phablet) {
        width = Breakpoint.Phablet - 1;
    } else {
        width = Breakpoint.Desktop + 1;
    }

    Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: width,
    });
};
