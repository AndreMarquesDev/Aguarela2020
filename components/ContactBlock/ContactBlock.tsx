import { useRouter } from 'next/router';
import React, { FC, useContext } from 'react';
import { Locale } from '../../utils/locales';
import TextsContext from '../context/TextsContext';
import TextShadowBlock from '../TextShadowBlock/TextShadowBlock';

const ContactBlock: FC = () => {
    const { texts } = useContext(TextsContext);

    const { query } = useRouter();
    const currentLanguage = query.language?.toString() as Locale;
    const titleMobile =
        currentLanguage === 'en'
            ? `${texts.contactMe1}${texts.contactMe2}t`
            : `${texts.contactMe1}${texts.contactMe2}${texts.contactMe3}`;

    return (
        <TextShadowBlock
            hasButton
            textBlock1={texts.needHelpWithYourBusiness}
            textBlock2={texts.sendMeAnEmail}
            textBlock3={texts.iAmAvailableToAdvise}
            title1={texts.contactMe1}
            title2={texts.contactMe2}
            title3={texts.contactMe3}
            titleMobile={titleMobile}
        />
    );
};

export default ContactBlock;
