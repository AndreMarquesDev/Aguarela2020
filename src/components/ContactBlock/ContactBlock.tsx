import type { Locale } from '../../types/Locale';
import { useRouter } from 'next/router';
import React, { useContext } from 'react';
import { contactBlockSectionDataTestId } from '../../utils/dataTestIds';
import { TextsContext } from '../context/TextsContext';
import { TextShadowBlock } from '../TextShadowBlock/TextShadowBlock';

export const ContactBlock = (): JSX.Element => {
    const { texts } = useContext(TextsContext);

    const { query } = useRouter();
    const currentLanguage = query.language?.toString() as Locale;
    const titleMobile
        = currentLanguage === 'en'
            ? `${texts.contactMe1}${texts.contactMe2}t`
            : `${texts.contactMe1}${texts.contactMe2}${texts.contactMe3}`;

    return (
        <TextShadowBlock
            hasButton
            dataTestId={contactBlockSectionDataTestId}
            textBlock1={texts.needHelpWithYourBusiness}
            textBlock2={texts.sendMeAnEmail}
            textBlock3={texts.iAmAvailableToAdvise}
            title1={texts.contactMe1}
            title2={texts.contactMe2}
            title3={texts.contactMe3}
            titleMobile={titleMobile}
        />
    );
};
