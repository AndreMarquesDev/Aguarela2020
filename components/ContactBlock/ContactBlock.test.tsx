import '@testing-library/jest-dom';
import React from 'react';
import { render, screen } from '@testing-library/react';
import * as nextRouter from 'next/router';
import ContactBlock from './ContactBlock';
import TextsContext from '../context/TextsContext';
import { textsEn, textsPt } from '../../utils/texts';
import { setJestWindowWidth } from '../../utils/jest/setJestWindowWidth';
import { Breakpoint } from '../../utils/useWindowSize';

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
nextRouter.useRouter = jest.fn(() => ({
    query: {
        language: 'pt',
    },
}));

describe('<ContactBlock />', () => {
    test('renders properly', () => {
        const { container } = render(<ContactBlock />);

        expect(screen.getByText(textsPt.contactMe1)).toBeInTheDocument();
        expect(screen.getByText(textsPt.contactMe2)).toBeInTheDocument();
        expect(screen.getByText(textsPt.contactMe3)).toBeInTheDocument();
        expect(screen.getByText(textsPt.needHelpWithYourBusiness)).toBeInTheDocument();
        expect(screen.getByText(textsPt.sendMeAnEmail)).toBeInTheDocument();
        expect(screen.getByText(textsPt.iAmAvailableToAdvise)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly on mobile', () => {
        setJestWindowWidth(Breakpoint.Mobile);

        const { container } = render(<ContactBlock />);

        expect(screen.queryByText(textsPt.contactMe1)).not.toBeInTheDocument();
        expect(screen.queryByText(textsPt.contactMe2)).not.toBeInTheDocument();
        expect(screen.queryByText(textsPt.contactMe3)).not.toBeInTheDocument();
        expect(
            screen.getByText(`${textsPt.contactMe1}${textsPt.contactMe2}${textsPt.contactMe3}`)
        ).toBeInTheDocument();
        expect(screen.getByText(textsPt.needHelpWithYourBusiness)).toBeInTheDocument();
        expect(screen.getByText(textsPt.sendMeAnEmail)).toBeInTheDocument();
        expect(screen.getByText(textsPt.iAmAvailableToAdvise)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly in English', () => {
        setJestWindowWidth(Breakpoint.Desktop);

        const { container } = render(
            <TextsContext.Provider
                value={{
                    texts: textsEn,
                }}
            >
                <ContactBlock />
            </TextsContext.Provider>
        );

        expect(screen.getByText(textsEn.contactMe1)).toBeInTheDocument();
        expect(screen.getByText(textsEn.contactMe2)).toBeInTheDocument();
        expect(screen.getByText(textsEn.contactMe3)).toBeInTheDocument();
        expect(screen.getByText(textsEn.needHelpWithYourBusiness)).toBeInTheDocument();
        expect(screen.getByText(textsEn.sendMeAnEmail)).toBeInTheDocument();
        expect(screen.getByText(textsEn.iAmAvailableToAdvise)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly in English and mobile', () => {
        (nextRouter.useRouter as jest.Mock).mockImplementationOnce(() => ({
            query: {
                language: 'en',
            },
        }));

        setJestWindowWidth(Breakpoint.Mobile);

        const { container } = render(
            <TextsContext.Provider
                value={{
                    texts: textsEn,
                }}
            >
                <ContactBlock />
            </TextsContext.Provider>
        );

        expect(screen.queryByText(textsEn.contactMe1)).not.toBeInTheDocument();
        expect(screen.queryByText(textsEn.contactMe2)).not.toBeInTheDocument();
        expect(screen.queryByText(textsEn.contactMe3)).not.toBeInTheDocument();
        expect(screen.getByText(`${textsEn.contactMe1}${textsEn.contactMe2}t`)).toBeInTheDocument();
        expect(screen.getByText(textsEn.needHelpWithYourBusiness)).toBeInTheDocument();
        expect(screen.getByText(textsEn.sendMeAnEmail)).toBeInTheDocument();
        expect(screen.getByText(textsEn.iAmAvailableToAdvise)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
