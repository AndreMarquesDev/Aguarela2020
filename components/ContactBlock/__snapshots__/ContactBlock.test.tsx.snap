// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactBlock /> renders properly 1`] = `
<div>
  <section
    class="jsx-2290377079 container"
  >
    <div
      class="jsx-2290377079 wrapper genericMargins"
    >
      <div
        class="jsx-2290377079 titleBlock"
      >
        <strong
          class="jsx-2290377079"
        >
          Con
        </strong>
        <strong
          class="jsx-2290377079"
        >
          Ver
        </strong>
        <strong
          class="jsx-2290377079"
        >
          Sar
        </strong>
      </div>
      <div
        class="jsx-2290377079 textBlock"
      >
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Precisa de ajuda com o seu negócio, projeto ou marca?
        </p>
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Envie-me um email e vamos conversar. Podemos começar o quanto antes a definir uma estratégia para social media de acordo com os seus objetivos.
        </p>
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Estou ainda disponível para dar consultoria em social media e partilhar todos os passos e dicas de que precisa para cumprir os seus objetivos.
        </p>
        <div
          class="jsx-781296904 buttonWrapper"
        >
          <div
            class="jsx-781296904 buttonBackground"
          >
            <a
              class="jsx-781296904 button uppercase"
              href="mailto:catarina@aguareladigital.com"
              rel="noreferrer"
              target="_blank"
            >
              catarina@aguareladigital.com
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`<ContactBlock /> renders properly in English 1`] = `
<div>
  <section
    class="jsx-2290377079 container"
  >
    <div
      class="jsx-2290377079 wrapper genericMargins"
    >
      <div
        class="jsx-2290377079 titleBlock"
      >
        <strong
          class="jsx-2290377079"
        >
          Con
        </strong>
        <strong
          class="jsx-2290377079"
        >
          Tac
        </strong>
        <strong
          class="jsx-2290377079"
        >
          Tme
        </strong>
      </div>
      <div
        class="jsx-2290377079 textBlock"
      >
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Need help with your business, project or brand?
        </p>
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Send me an email and let's talk. We can start as soon as possible defining a social media strategy that meets your objectives.
        </p>
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          I am also available to provide social media consulting and share all the tips you need to fulfill your goals.
        </p>
        <div
          class="jsx-781296904 buttonWrapper"
        >
          <div
            class="jsx-781296904 buttonBackground"
          >
            <a
              class="jsx-781296904 button uppercase"
              href="mailto:catarina@aguareladigital.com"
              rel="noreferrer"
              target="_blank"
            >
              catarina@aguareladigital.com
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`<ContactBlock /> renders properly in English and mobile 1`] = `
<div>
  <section
    class="jsx-2290377079 container"
  >
    <div
      class="jsx-2290377079 wrapper genericMargins"
    >
      <div
        class="jsx-2290377079 titleBlock"
      >
        <p
          class="jsx-2290377079"
        >
          ConTact
        </p>
      </div>
      <div
        class="jsx-2290377079 textBlock"
      >
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Need help with your business, project or brand?
        </p>
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Send me an email and let's talk. We can start as soon as possible defining a social media strategy that meets your objectives.
        </p>
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          I am also available to provide social media consulting and share all the tips you need to fulfill your goals.
        </p>
        <div
          class="jsx-781296904 buttonWrapper"
        >
          <div
            class="jsx-781296904 buttonBackground alignCenter"
          >
            <a
              class="jsx-781296904 button uppercase"
              href="mailto:catarina@aguareladigital.com"
              rel="noreferrer"
              target="_blank"
            >
              catarina@aguareladigital.com
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`<ContactBlock /> renders properly on mobile 1`] = `
<div>
  <section
    class="jsx-2290377079 container"
  >
    <div
      class="jsx-2290377079 wrapper genericMargins"
    >
      <div
        class="jsx-2290377079 titleBlock"
      >
        <p
          class="jsx-2290377079"
        >
          ConVerSar
        </p>
      </div>
      <div
        class="jsx-2290377079 textBlock"
      >
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Precisa de ajuda com o seu negócio, projeto ou marca?
        </p>
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Envie-me um email e vamos conversar. Podemos começar o quanto antes a definir uma estratégia para social media de acordo com os seus objetivos.
        </p>
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Estou ainda disponível para dar consultoria em social media e partilhar todos os passos e dicas de que precisa para cumprir os seus objetivos.
        </p>
        <div
          class="jsx-781296904 buttonWrapper"
        >
          <div
            class="jsx-781296904 buttonBackground alignCenter"
          >
            <a
              class="jsx-781296904 button uppercase"
              href="mailto:catarina@aguareladigital.com"
              rel="noreferrer"
              target="_blank"
            >
              catarina@aguareladigital.com
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
